cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -220,4 +256,83 @@ startLedgerNewEpochStateLogging testnetRuntime tmpWorkspace = withFrozenCallStac
             <> displayException e <> "\n"
           pure ConditionMet
 
+-- TODO: I hate this. Is there a better way to do this without
+-- reaching for concurrency primitives?
+isFileReadableLoop :: FilePath -> IO Bool
+isFileReadableLoop fp = do
```

Comment:
`appendFile`s in 248 and 249 are constantly opening and closing the files so I don't think this will work - I think you might get inbetween accidentally. I second John's advice https://github.com/IntersectMBO/cardano-node/pull/5854/files#r1611865524